File: src/common/actor/ILevel.java
package common.actor;

import javafx.scene.canvas.Canvas;
import javafx.scene.input.KeyEvent;

public interface ILevel {
    /**
     * Create all your level elements
     * @param gameCanvas
     */
    void createLevel( Canvas gameCanvas );

    /**
     * What happens, when a key is pressed?
     */
    void keyboardInput( KeyEvent keyEvent );

    /**
     * Draw what you would like to :)
     */
    void render( Canvas canvas, final int fps );

}

File: Common/Actor/src/common/actor/ILevel.kt
package common.actor

import javafx.scene.canvas.Canvas
import javafx.scene.input.KeyEvent

interface ILevel {
    /**
     * Create all your level elements
     * @param gameCanvas
     */
    fun createLevel(gameCanvas: Canvas?)

    /**
     * What happens, when a key is pressed?
     */
    fun keyboardInput(keyEvent: KeyEvent?)

    /**
     * Draw what you would like to :)
     */
    fun render(canvas: Canvas?, fps: Int)
}
